arango/batch.py

```diff
@@ -225,6 +284,8 @@ def __init__(self):
         self._status = 'pending'
         self._result = None
 
+        self._lock = Lock()
```

Comment:
How come we are not doing the python2 vs 3 check here?

Reply:
It doesn't need a timeout.  Python 2 has Lock, it just doesnt have a timeout

Reply:
Got it.